src/dune/exe.mli

```diff
@@ -41,6 +41,7 @@ val build_and_link :
   -> promote:Rule.Promote.t option
   -> ?link_args:Command.Args.static Command.Args.t Build.t
   -> ?o_files:Path.t list
+  -> ?custom_build_info:Custom_build_info.t option
```

Comment:
The optional option argument is a bit confusing. Since there's no difference between passing None and not passing the argument I would recommend either making the argument mandatory or turning it into `?custom_build_info:Custom_build_info.t`.